src/common.ts

```diff
@@ -103,15 +115,25 @@ export abstract class Kommand extends Command {
         await this.sdk.init(this)
       }
 
-      if (this.flags.as) {
-        this.logInfo(`Impersonate user "${this.flags.as}"`)
-        await this.sdk.impersonate(this.flags.as, async () => {
-          await this.runSafe()
-        })
+      const analytics = async () => {
+        try {
+          return await this.analytics.track(kommand.id, this.config.name, this.config.version);
+        } catch (_) { /* We should not interfere with the process exit code */ }
       }
-      else {
-        await this.runSafe()
+
+      const runCmd = () => {
+        if (this.flags.as) {
+          this.logInfo(`Impersonate user "${this.flags.as}"`)
+          return this.sdk.impersonate(this.flags.as, async () => {
+            return this.runSafe()
+          })
+        }
+        else {
+          return this.runSafe()
+        }
       }
+
+      await Promise.all([analytics(), runCmd()]);
```

Comment:
The `analytics` promise should not be awaited to not interfere with Kourou usage